src/resolver.rs

```diff
@@ -1353,7 +1353,38 @@ impl<'i> TypeAnnotator<'i> {
                                     .or_insert_with(std::vec::Vec::new)
                                     .push(candidate.to_string())
                             } else {
-                                params.push((parameter, type_name.to_string()))
+                                // intrinsic type promotion for variadics in order to be compatible with the C standard.
```

Comment:
first of all - wow - great job investigating what the problem might be and even finding such a great resource that explains the behavior! :+1: 
can you add a comment that explains the else-situation?
i think it is something like, variadic-parameter without type-declaration?

Reply:
I can't really take all the credit here. Had a discussion with Ghaith about it, and also we had another issue where printf would print any f32 value as 0. This was a pretty good indicator that something was wrong with the parameter types, but thank you anyway!